task/ko-oci-ta/0.1/ko-oci-ta.yaml

```diff
@@ -0,0 +1,157 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: ko-oci-ta
+  annotations:
+    tekton.dev/categories: Image Build
+    tekton.dev/displayName: Build and upload container image using ko oci
+      trusted artifacts
+    tekton.dev/pipelines.minVersion: 0.17.0
+    tekton.dev/platforms: linux/amd64,linux/arm64
+    tekton.dev/tags: image-build
+  labels:
+    app.kubernetes.io/version: "0.1"
+spec:
+  description: This Task builds source into a container image using ko.
+  params:
+    - name: BUILD_ARGS
+      description: additional arguments to ko build
+      type: string
+      default: ""
+    - name: CACHI2_ARTIFACT
+      description: The Trusted Artifact URI pointing to the artifact with
+        the prefetched dependencies.
+      type: string
+      default: ""
+    - name: COMMIT_SHA
+      description: The image is built from this commit.
+      type: string
+      default: ""
+    - name: IMAGE
+      description: Reference of the image ko will produce.
```

Comment:
Based on what I learned in https://github.com/konflux-ci/build-definitions/pull/2670#discussion_r2410450678, I think this parameter is misleading. If I set it to `quay.io/my-org/my-image:latest`, I would expect the task to push there, but it will instead push to `quay.io/my-org/<name of binary>:latest`

Maybe the task should verify that the repo name from `$IMAGE` matches the name of the binary and fail otherwise?

Reply:
This probably isn't needed. Instead, I think that `KO_DOCKER_REPO` should be passed in. With that and the `IMPORT_PATH` we should have everything we need.

I am also assuming `--base-import-paths` would be hard-coded. Does this all sound OK to you, @vdemeester?